usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -68,11 +69,16 @@ def search_transactions(filters, fields, sort, order, lower_limit, limit):
     index_name = '{}-{}'.format(TRANSACTIONS_INDEX_ROOT,
                                 transaction_type.lower().replace(' ', ''))
     index_name = index_name[:-1]+'*'
-    try:
-        response = CLIENT.search(index=index_name, body=query)
-    except Exception:
-        logging.exception("There was an error connecting to the ElasticSearch instance.")
-        return None
+    found_result = 0
+    while not found_result > 10:
+        found_result += 1
+        try:
+            response = CLIENT.search(index=index_name, body=query)
+        except (TransportError, ConnectionError) as e:
+            logger.error(e)
+            logging.exception("There was an error connecting to the ElasticSearch instance.")
+            if found_result == 10:
+                return HttpResponseServerError()
```

Comment:
What is the behavior of using this return object for the functions? It might work for the Django view classes but I am skeptical it works for these helper functions. Can you please share in the PR comments before and after responses showing the differences?

Reply:
When there is no connection I get this:

Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa198>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa198>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa668>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa668>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa630>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa630>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa2b0>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa2b0>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa7f0>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa7f0>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aaeb8>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aaeb8>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aac88>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aac88>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa978>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa978>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa278>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa278>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
14:21:31 - ConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa550>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1127aa550>: Failed to establish a new connection: [Errno 61] Connection refused)
14:21:31 - Error retrieving sums and counts. Retrying connection.
[09/Feb/2018 14:21:31] "POST /api/v2/search/transaction_spending_summary HTTP/1.1" 500 27

So it tried 9-10 times, tells me I get a Connection error, gives the logger error, and then returns a 500 error after those 9-10 times.

Reply:
I understand that part. What I don't understand is how the calling functions handle the HttpResponseServerError object as a return value

Reply:
<h1>Server Error (500)</h1> is what I'm getting as a response value.

Reply:
Handles the same when I put it directly in the post function. Still returns Server Error (500).

class TransactionSummaryVisualizationViewSet(APIView):

    @cache_response()
    def post(self, request):
        """
            Returns a summary of transactions which match the award search filter
                Desired values:
                    total number of transactions `award_count`
                    The federal_action_obligation sum of all those transactions `award_spending`

            *Note* Only deals with prime awards, future plans to include sub-awards.
        """
        json_request = request.data
        filters = json_request.get("filters", None)
        if filters is None:
            raise InvalidParameterException("Missing one or more required request parameters: filters")

        results = HttpResponseServerError()
        # build response
        return Response({"results": results})

Reply:
My point I'm trying to get at is that the final 500 response isn't generated by HttpResponseServerError() but by the calling function in `search.py` is trying to use that object incorrectly and creating a python exception there which is the real 500 error. Could you put a message string in the HttpResponseServerError object and see if that is returned?

Reply:
Ah, I see. No the message string is not returned.